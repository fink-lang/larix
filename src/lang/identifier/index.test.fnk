{describe, it, expect, to_match_snapshot, to_throw} = import '@fink/jest'

{parse_expr, parse} = import '../../'


describe:: 'other symbols', fn:
  it:: 'parses a word', fn:
    expect::
      parse_expr:: `  foobar  `
      to_match_snapshot


  it:: 'parses a non-word', fn:
    expect::
      parse_expr:: `  Ï€  `
      to_match_snapshot


describe:: 'symbols as prefix operators', fn:
  it:: 'parses with left to right precendence', fn:
    expect::
      parse:: `
        foo bar, spam, shrub
        foo bar, spam shrub
        foo
          bar
          spam
            1 + 23
            234
          ni(na)
        foo bar,
          fn bar:
            spam
          ni
            na
      `
      to_match_snapshot
