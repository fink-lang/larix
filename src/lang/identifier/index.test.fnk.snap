// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`other symbols parses a non-word 1`] = `ident (1:2-1:3) Ï€`;

exports[`other symbols parses a word 1`] = `ident (1:2-1:8) foobar`;

exports[`symbols as prefix operators parses with left to right precendence 1`] = `
module (1:0-14:0)
  call (1:0-1:20)
    ident (1:0-1:3) foo
    ident (1:4-1:7) bar
    ident (1:9-1:13) spam
    ident (1:15-1:20) shrub
  call (2:0-2:19)
    ident (2:0-2:3) foo
    ident (2:4-2:7) bar
    call (2:9-2:19)
      ident (2:9-2:13) spam
      ident (2:14-2:19) shrub
  call (3:0-8:8)
    ident (3:0-3:3) foo
    ident (4:2-4:5) bar
    call (5:2-7:7)
      ident (5:2-5:6) spam
      arithm + (6:4-6:10)
        number (6:4-6:5) 1
        number (6:8-6:10) 23
      number (7:4-7:7) 234
    call (8:2-8:8)
      ident (8:2-8:4) ni
      ident (8:5-8:7) na
  call (9:0-14:0)
    ident (9:0-9:3) foo
    ident (9:4-9:7) bar
    block fn (10:2-11:8)
      ident (10:5-10:8) bar
      :
      ident (11:4-11:8) spam
    call (12:2-14:0)
      ident (12:2-12:4) ni
      ident (13:4-13:6) na
`;
